var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -108,6 +108,8 @@ class Hdf5(CMakePackage):
     conflicts('+java', when='@:1.9')
     # The Java wrappers cannot be built without shared libs.
     conflicts('+java', when='~shared')
+    # Fortran cannot be built with shared
+    conflicts('+fortran', when='+shared@:1.8')
```

Comment:
I'm a little confused about this change.  We've been using hdf5@1.8+fortran+shared for a while without any issues, and a number of users depend on it. What problem does this avoid?

Reply:
Hm, I'll have to revisit my build logs, but I got a configure or build error with this configuration -- but it might have also been `+cxx` which I neglected here...

Reply:
FWIW this configuration works for us: `hdf5@1.8.22%gcc@7.5.0+cxx+fortran+hl~ipo~java~mpi+shared~szip~threadsafe+tools api=default build_type=RelWithDebInfo arch=linux-centos7-broadwell` as well as +mpi and %gcc@10.

Reply:
I just tried two different builds myself and couldn't reproduce the failure. Sorry about that and thanks for commenting—I'll put in a PR to revert this now.

Reply:
Oh derp, I recorded the failure in my commit history -- for some reason it had concretized to an old 1.8.15, which I assumed was the newest of the .8 series.

Reply:
https://github.com/spack/spack/pull/29472